script.js

```diff
@@ -0,0 +1,100 @@
+// CHALLENGE 1
+// `// given an array of post objects create a function that returns a new array of posts, each object in the new array should have keys renamed as per the example. Original array of posts should not be mutated.`
+
+// `// @param {{id: number, createdBy: number, commentNo: number}[]}`
+
+// `// @returns {{postId: number, postCreatedBy: number }[]}`
+
+const postsArray = [
+  {
+    id: 123,
+    createdBy: "user16",
+    commentNo: 5,
+  },
+  {
+    id: 456,
+    createdBy: "user20",
+    commentNo: 5,
+  },
+  {
+    id: 789,
+    createdBy: "user32",
+    commentNo: 5,
+  },
+];
+
+const newPostArr = (arr) => {
+  const newArr = arr.map((item) => {
+    return {
+      postsId: item.id,
+      postsCreated: item.createdBy,
+    };
+  });
+  console.log(newArr);
+};
+
+newPostArr(postsArray);
+
+// Challenges using the SWAPI API: https://swapi.dev/
+
+// CHALLENGE 2
+// Fetch all people from the API, create a function that takes in the results array and returns a new array only with people with height above 166
+
+const fetchData = async (searchTerm) => {
+  const response = await fetch(`https://swapi.dev/api/${searchTerm}/`);
+
+  const data = await response.json();
+  return data.results;
+};
+
+const heightFilter = async () => {
+  const people = await fetchData("people");
+
+  const newArr = people.filter((item) => {
+    return item.height > 166;
+  });
+  return newArr;
+};
+
+heightFilter();
+
+// CHALLENGE 3
+// Fetch all starships from the API. Create a function that takes in a the results array and returns a new array with all starships, but only the name, model and people property for each object.
+
+const starShips = async () => {
+  const ships = await fetchData("starships");
+  const newArr = ships.map((ship) => {
+    return { name: ship.name, model: ship.model, pilots: ship.pilots };
```

Comment:
as an alternative it could be done with object destructuring:

```js
const newArr = ships.map((ship) => {
    const {name, model, pilot, ...rest} = ship
    return { name, model, pilots };
```